### PR TITLE
 #75526 Provisioning service: Prepare the new service interface - VA-119

### PR DESCRIPTION
**Pull request by ctthang**

### CODE REVIEW

File: interface/CH.json [hunk 1]
```diff
@@ -1669,21 +1439,16 @@
           {
             "name": "id",
             "in": "path",
-            "description": "The id of the user system to add SAML2.0 metadata version for. This is the SAML2.0 entityId of the BrugervendtSystem. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
+            "description": "The id of the user system to set OIO SAML profile for. This is the SAML2.0 entityId of the BrugervendtSystem. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
```
Comment (thuannguy) [inline interface/CH.json]: OIOSAML, not OIO SAML

File: interface/CH.json [hunk 2]
```diff
@@ -1652,10 +1422,10 @@
         }
       }
     },
-    "/BrugervendteSystemer/{id}/SamlMetadata": {
+    "/BrugervendteSystemer/{id}/OIOSAMLProfile/{value}": {
       "put": {
-        "summary": "Add a SAML metadata version on the specified user system",
-        "description": "Add a `SamlMetadata` version on the specified `BrugervendtSystem`",
+        "summary": "Set the OIOSAMLProfile for this BrugervendtSystem (user system)",
+        "description": "Set the 'OIOSAMLProfile' for this `BrugervendtSystem` (user system)",
```
Comment (thuannguy) [inline interface/CH.json]: 'OIOSAMLProfile' -> should this one be `OIOSAMLProfile`? Many other places have the same mistake as well

File: interface/CH.json [hunk 3]
```diff
@@ -1729,32 +1494,109 @@
...
+        "consumes": [
+          "application/json"
+        ],
         "parameters": [
           {
-            "name": "id1",
+            "name": "id",
             "in": "path",
-            "description": "The id of the user system to delete a SAML2.0 metadata version for. This is the SAML2.0 entityId of the BrugervendtSystem. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
+            "description": "The id of the user system to set NSIS Assurance Level for. This is the SAML2.0 entityId of the BrugervendtSystem. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
```
Comment (thuannguy) [inline interface/CH.json]: to set *a* NSIS Assurance Level for

File: interface/CH.json [hunk 4]
```diff
@@ -2403,35 +2245,79 @@
...
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
-            "name": "id1",
+            "name": "id",
             "in": "path",
-            "description": "The id of the IdentityProvider. This is the SAML2.0 entityId of the IdentityProvider. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
+            "description": "The id of the identity provider to set mobile friendly for. This is the SAML2.0 entityId of the IdentityProvider. The id must be the base64 encoded. See processing instructions for handling base64 encoding, to ensure interoperability.",
```
Comment (thuannguy) [inline interface/CH.json]: mobile friendly? Seem a wrong description

**Review by thuannguy — CHANGES_REQUESTED** [review]
As-is, an error response from the provisioning service has two attributes: a Code and a Message. The new provisioning service will have a new Version attribute that is set to “2.0” to indicate that a response is from the new service. The old service is kept as-is.

File: interface/CH.json (hunk 1 above)
Comment (ctthang) [reply interface/CH.json]: Fixed

File: interface/CH.json (hunk 2 above)
Comment (ctthang) [reply interface/CH.json]: fixed

File: interface/CH.json (hunk 3 above)
Comment (ctthang) [reply interface/CH.json]: fixed

File: interface/CH.json (hunk 4 above)
Comment (ctthang) [reply interface/CH.json]: yes, fixed

**Comment by ctthang** [comment]
> As-is, an error response from the provisioning service has two attributes: a Code and a Message. The new provisioning service will have a new Version attribute that is set to “2.0” to indicate that a response is from the new service. The old service is kept as-is.

Added "version" attribute to the Error model

**Review by thuannguy — APPROVED** [review]

**Pull request merged**